btree/btree.c

```diff
@@ -1986,22 +2104,57 @@ static void ff_del(const struct nd *node, int idx, struct m0_be_tx *tx)
 	h->ff_used--;
 }
 
+static void ff_set_level(const struct nd *node, uint8_t new_level)
+{
+	struct ff_head *h = ff_data(node);
+
+	h->ff_level = new_level;
+}
+
 static void generic_move(struct nd *src, struct nd *tgt,
 			 enum dir dir, int nr, struct m0_be_tx *tx)
 {
+	struct slot rec;
+	struct slot tmp;
+	m0_bcount_t rec_ksize;
+	m0_bcount_t rec_vsize;
+	m0_bcount_t temp_ksize;
+	m0_bcount_t temp_vsize;
+	void       *rec_p_key;
+	void       *rec_p_val;
+	void       *temp_p_key;
+	void       *temp_p_val;
+	int         srcidx;
+	int         tgtidx;
+	int         last_idx_src;
+	int         last_idx_tgt;
+
+	rec.s_rec.r_key.k_data.ov_vec.v_count = &rec_ksize;
+	rec.s_rec.r_key.k_data.ov_buf = &rec_p_key;
+	rec.s_rec.r_val.ov_vec.v_count = &rec_vsize;
+	rec.s_rec.r_val.ov_buf = &rec_p_val;
+
+	tmp.s_rec.r_key.k_data.ov_vec.v_count = &temp_ksize;
+	tmp.s_rec.r_key.k_data.ov_buf = &temp_p_key;
+	tmp.s_rec.r_val.ov_vec.v_count = &temp_vsize;
+	tmp.s_rec.r_val.ov_buf = &temp_p_val;
```

Comment:
` ov_vec and ov_buf ` intialisation code is repetitive. Can add a common function to initialise these values.

Reply:
there is M0_BUFVEC_INIT_BUF(); I can use that for initialization